### ISSUE TITLE
Releasing the app by signing doesn't work with *.jks file. 

### ISSUE THREAD

**Issue by NickIliev** [NativeScript/sidekick-feedback#119]
_From @Sharique-Hasan on January 15, 2018 23:19_

Hello, 

I had a native app already published in playstore.Now I have ported it on nativescript with angular. To release the app I have to sign it with the same key. When I tried to do that I got an error

`The keystore is tempered or the password is incorrect. `

However I am providing valid keystore and password. I tested this with **Sidekick**.
- Created a new app with it. 
- Built the release version of app with same keystore and password

Project builds successfully. Somehow it is rejecting the *.jks keystore on commandline. When I signed it with newly created keystore *.keystore it built the app successfully. What am I missing with *.jks keystore signing? 

_Copied from original issue: NativeScript/nativescript-cli#3320_

**Comment by Sharique-Hasan** [comment]
@NickIliev but it is related to cli. As the supposed behavior is working with Sidekick but failing in CLI

**Comment by NickIliev** [comment]
@Sharique-Hasan I am not able to reproduce the issue while testing a release build in CLI with provided *.jks file. Can you clarify if you have tried building in release with CLI commands or with Sidekick (in the initial post you are saying that the tests are made with Sidekick)

**Comment by Sharique-Hasan** [comment]
@NickIliev 

I have to use my previous native android's keystore and use it for webpack.

```npm run build-android-bundle -- --env.production --release --keyStorePath ./keystore/oldAndroid.jks --keyStorePassword password --keyStoreAlias androidAlias --keyStoreAliasPassword password```

It fails with the above error.

When I use this command:

```npm run build-android-bundle --uglify -—snapshot -- --env.production --release --keyStorePath ./keystore/newNsAndroid.keystore --keyStorePassword password --keyStoreAlias newAndroidAlias --keyStoreAliasPassword password```

Builds successfully. 


When I used *.jks with Sidekick. Sidekick built the app successfully. Build with *.jks keystore fails only on CLI. If you need I can provide my app and keystore privately and you can reproduce it.

**Comment by NickIliev** [comment]
@Sharique-Hasan I've noticed that you are using 'npm run ...' while the latest version of NativeScript and nativescript-dev-webpack has introduced a different workflow with CLI commands. If possible please [migrate your application](https://www.nativescript.org/blog/nativescript-webpack-0.9.0-what-changed-and-how-to-upgrade) and try building in release again using 
```
tns run android --bundle --env.uglify --release --keyStorePath ./keystore/oldAndroid.jks --keyStorePassword password --keyStoreAlias androidAlias --keyStoreAliasPassword password`.
```

**Comment by Sharique-Hasan** [comment]
Yes my current project is standing at NS 3.2.0. I tried migrating the project. there were few issues with the plugins I used. So I have a deadline to release my app on playstore this weekend. Therefore I am compelled to release the app with 3.2.0 as I have a Go-A-head from my QA department. Migrating would require another cycle of testing. I hope this make sense

**Comment by Sharique-Hasan** [comment]
Hey nick. Any update over this? I am actually getting little late from my deadline. If the sidekick built the app using webpack then I am fine with that. But I guess it doesn't. Please update about this issue.

**Comment by Sharique-Hasan** [comment]
UPDATE:

@tsvetie  @NickIliev I checked the app with {N}v3.4.1 and the error was same.

**Comment by NickIliev** [comment]
@Sharique-Hasan hard to say what might be causing the issue as it is not reproducible on my side.
However, differences in the Java version and Gradle plugin might cause some known bugs - are you by change using different passwords for your alias and store pass? If so check out this solution [here](https://stackoverflow.com/a/33071940/4936697). 
There are a number of reasons that can cause `keystore is tempered or the password is incorrect` so you might want to check the most reported ones as they are usually connected to the keytool file itself.

**Comment by Sharique-Hasan** [comment]
I tested few experiments. I found out that CLI is not accepting the
keystore password with special characters. I had `$` and it turned it into
some numerics. Therefore I guess the wrong password error was coming up.
You can test it at your end too.

**Comment by NickIliev** [comment]
@Sharique-Hasan good to hear that you have found the reason for this issue!
Meanwhile, we have not released version 1.2.0 of Sidekick which can now supports build with Webpack so if you still need to resolve this one you can try building it with the latest Sidekick by enabling the Webpack (install nativescript-dev-webpack in Sidekick to get access to the Webpack build)

**Comment by NickIliev** [comment]
@Sharique-Hasan second thought about using special symbols like ``$` in your terminal. 
We should use escaping when entering these symbols (Sidekick is escaping them out-of-the-box which was the reason why your password was not accepted in CLI but accepted in Sidekick)

For example see the difference between:
```
echo $PATH
```
and the same with escaping 
```
echo \$PATH
```
